packages/react-form/src/hooks/field/field.ts

```diff
@@ -245,6 +245,41 @@ export function asChoiceField<Value>(
   };
 }
 
+/**
+ * Converts a standard `Field<Value>` into a form that is compatible
+ * with the `<ChoiceList />` component in `@shopify/polaris`.
+ *
+ * For fields that are used by both choice components and other components, it
+ * can be beneficial to retain the original `Field<Value>` shape and convert
+ * the field on the fly for the choice component.
+ *
+ * ```typescript
+ * enum Color { Red = "red", Green = "green" }
+ *
+ * const choices = [
+ *   { label: "Red", value: Color.Red },
+ *   { label: "Green", value: Color.Green },
+ * ]
+ *
+ * const color = useField(Color.Red);
+ * return (<ChoiceList {...asChoiceList(color)} label="Color" choices={choices} />);
+ * ```
+ */
+export function asChoiceList<Value extends string | undefined | null>({
+  value,
+  onChange,
+  ...fieldData
+}: Field<Value>) {
+  return {
+    ...fieldData,
+    // eslint-disable-next-line eqeqeq
+    selected: value == undefined ? [] : [value!],
+    onChange: (selected: string[]) => {
+      onChange(selected[0]! as Value);
```

Comment:
This is an interesting point, it would only work for radio buttons, not checklists. I think we would need a separate handler for converting useList to ChoiceList.

How do you want to proceed here? We could maybe detect what's passed in and react accordingly?

Reply:
Actually it looks like `useList` and `useDynamicList` don't allow lists of strings, only lists of objects. So they aren't usable here.

Reply:
Yeah `useList` is a bit more flexible, it expects an object of various field types.
I think the most straightforward thing to do for now would be to have `asChoiceField` provide `allowMultiple: false` in it's return value and explicitly document that it doesn't work with checkboxes yet, we can always add that later.